clients/src/main/java/org/apache/kafka/common/network/NetworkReceive.java

```diff
@@ -146,6 +147,12 @@ public ByteBuffer payload() {
         return this.buffer;
     }
 
+    public int bytesRead() {
+        if (buffer == null)
+            return size.position();
+        return payload().position() + size.position();
```

Comment:
Looks odd to use `buffer` a couple of lines above and `payload()` here since both refer to the same thing.